eglot-jl.jl

```diff
@@ -1,13 +1,33 @@
-# Make sure that we only load packages from this environment specifically.
+# Usage:
+#   julia --project path/to/eglot-jl/eglot-jl.jl [DEPOT_PATH]
+
+# Get the project environment.
+#
+# WARNING: this script must be called with the `--project` command-line switch
+# for this to work reliably.
+project_path = dirname(Base.load_path()[1])
```

Comment:
Since `Base.load_path` is an unexported function anyway, let's just directly use `Base.current_project()` and not call this script with `--project`. That way we're only relying on a non-public API rather than relying on a non-public API plus breaking the semantics of `--project`.

That can also avoids the usage of `Pkg.activate` which is intended to be more for interactive use.

Reply:
This is a good idea. Although it won't be that simple: `Base.load_path` returns `nothing` for directories that are outside of any project.

The correct logic to get the project associated to any directory would be along the lines of:
```
find_project(dir) = something(Base.current_project(dir), Base.load_path_expand(LOAD_PATH[2]))
```

I'll try and propose a new commit implementing this.